### ISSUE TITLE
How can I create room mailboxes programmatically with EWS ?

### ISSUE THREAD

**Issue by ststeiger** [OfficeDev/ews-managed-api#189]
We have a facility management application, that manages all rooms in a large corporation. 
Our customers has the bad idea that he wants to be able to manage rooms via Outlook (exchange meeting requests), although our software already has a web-ui for this. 

Problem is, apparently exchange needs a room mailbox for every room. 
So, if we currently have 20'000 rooms in the test-system, and 18'977 rooms in the prod-system, how can we automatically create rooms in exchange from the web-application (if a new building is being requisitioned or if a room is cut into half) ?  
(and also, in case 2 rooms are merged into one, how to remove rooms)

As far as I can tell, EWS doesn't support this use-case. 
What would be the best way to integrate this into a web-application ? 

I presume other people must have had the same problem before me...
Certainly, manually inputing every record can't be the solution, and neither can't a powersell-cmdlet...

**Comment by MichelZ** [comment]
I think you have to resort to PowerShell.
The good news is that you can use PowerShell from C# code...

**Comment by ststeiger** [comment]
@MichelZ: Yea, the bad news is on the exchange server, not the web-server...

**Comment by MichelZ** [comment]
What do you mean? I regularly use PS with Exchange from non-exchange servers.... 
all you need is the Exchange Management Shell installed on the Webserver

**Comment by ststeiger** [comment]
Ah, ok, well, in that case, it possibly works, but that still sucks.

**Comment by MichelZ** [comment]
Yes, a proper API would be nicer, this is definitely true. Maybe Exchange 2019 gets it, who knows :)

**Comment by ststeiger** [comment]
Wibder uf Exchange Management Shell works on Linux with wine ;)

**Comment by davster** [comment]
EWS does not support management operations such as creating or deleting mailboxes.  Remote powershell is likely your best option there.

**Comment by ststeiger** [comment]
Erm, thinking about it. 
Does it even work on Windows ? 

Say I install Exchange Management Shell on the Webserver. 
But the webserver runs in one of our data-center servers, inside our datacenter-domain. 
The exchange installation runs in the client's domain, which is NOT in the datacenter. 
Does Exchange Management Shell work under this conditions ?

**Comment by davster** [comment]
O365 has a powershell virtual directory just like EWS or REST.  In fact, even when running powershell from within the O365 server environment it is making "remote" powershell calls.

This may help:

https://docs.microsoft.com/en-us/powershell/exchange/exchange-online/connect-to-exchange-online-powershell/connect-to-exchange-online-powershell?view=exchange-ps
